locator/token_metadata.cc

```diff
@@ -50,9 +50,9 @@ class token_metadata_impl final {
      */
     // FIXME: have to be BiMultiValMap
     std::unordered_map<token, inet_address> _token_to_endpoint_map;
-    // Track the number of nodes in _token_to_endpoint_map. Need to update
-    // _nr_normal_token_owners when _token_to_endpoint_map is updated.
-    size_t _nr_normal_token_owners;
+
+    // Track the unique set of nodes in _token_to_endpoint_map
+    std::unordered_set<inet_address> _normal_token_owners;
```

Comment:
Some things we can consider (not proposing for now):
 - switch to abseil btree containers for fewer allocations/branches
 - store all node vectors in a few larger vector and return std::span<>s to it
 - start storing node* instead of inet_address, pointing to a vector<node> i token_metadata

The last two require that we hold token_metadata_ptr in the read/write executors, but that's easy.

Reply:
I think that the node instances belong elsewhere, not in token_metadata, maybe in topology.
and then they can be reference by a shared ptr until they are removed from the cluster.